### DISCUSSION TITLE
Running AG as dual mono plugin? (LPX)

### DISCUSSION THREAD

**Discussion by dsponaugle** [apohl79/audiogridder#365] [General]
If I load up an AG instance in LPX as dual-mono (as opposed to mono or stereo), I'm supposed to be able to couple the l/r channels and they are then processed independently. Any adjustments are duplicated as if 2 AG instances were loaded. This does not seem to work with AG. Is this a possibility or an LPX limitation? I can set up 2 AG instances and map to smart controls, but the smart controls are limited to 16 knobs and 16 switches at most and it's a very time-consuming adventure. This would be a much quicker way, for example to process double-tracked & hard-panned guitars in a summing track stack without all of the smart control setup since you could adjust one instance of AG as dual mono. I could see other use cases for this as well. Just throwing it out into the ether! Thanks for the incredible product!

**Comment by apohl79** [comment]
>  Is this a possibility or an LPX limitation? 

Thats a good question, I have never used this feature in logic and don't know how it is supposed to behave.

> but the smart controls are limited to 16 knobs and 16 switches

AG exposes 16 parameters per default, that you can map plugin parameters to for automation. You can change that number in the config file to whatever you need. (https://audiogridder.com/plugin/) - but I guess thats now what you are asking for...

> This would be a much quicker way

Still not sure, how AG should behave. I guess it should behave like a single plugin instance, so all changes on the L side are in sync with R and vice versa?

**Reply by dsponaugle** [reply]
To be honest, I had never used dual mono either until this scenario. I was trying to find a way to not have to map tons of smart controls just to run 2 virtual guitar amps (hard panned) and be able to adjust identical controls on both at the same time in LPX.  There is a feature in protools that I think should be in every DAW... and you can just group whatever changes are across multiple channels and it just works. Every parameter of a plugin will be duplicated across multiple tracks if need be. It keeps you in "the zone" when you're mixing so you don't have to mute tracks, pan center, make adjustments and then duplicate them to the other track. Smart controls is one way of achieving similar results to dual mono but it's quite limited and super time-consuming. After more research, it's actually a max of 24 knobs and 24 switches in smart controls but that's a lot of mapping to do. I just adjusted the AG's default automation limit in the config file  but then I'm still stuck to the smart controls maximums. 

Essentially, dual-mono actually spins up 2 plugins at once and you have the option of processing the left & right channels separately, or couple them together. If AG is loaded as dual mono (coupled option chosen) to a summing track stack, you could have 2 guitar tracks within the stack hard panned left/right and not have to automate any parameters at all. You just adjust the plugin and it's actually adjusting 2 copies of the plugin. Essentially the left channel and right channel are processed separately but taking the same settings since the dual mono is coupled. Both instruments feed into the summing track stack and you can real-time modify plugin parameters affecting both guitar tracks (as an example). Of course, there's still a need for automation in other use-cases, but AG doesn't seem to duplicate settings on the l/r side when coupled at all so there isn't any dual-mono integration even though it can load up as a dual mono plugin.

This would solve some major problems for people that dual-track instruments but want to adjust both channels together. IT's not the same as just throwing a stereo amp sim on a bus since both guitar tracks would be feeding to the same instance of the amp and fighting one another. dual mono is actually super cool for compressors as well since the audio on the left and right could be different between takes and it gives the compressor the ability to properly react to the differing peaks & valleys from each side. 

Sorry to ramble on, but dual-mono functionality could be super useful for people to offload cpu-hungry amp sims (or whatever you're using) to other machines but still have the same functionality as if they were running locally in dual-mono mode, thus avoiding the need to create lots of automation links and somewhat limited and time-consuming smart controls.
